### PR TITLE
New function time.ymd()

### PR DESCRIPTION
**Pull request by st-pasha**
Function `dt.time.ymd()` can be  used to create `date32` columns out of year/month/day components.

I've decided that all date/time functions will leave in the `datatable.time` namespace, so that they can be easily discovered  and run less chance of clashing with some other functionality.

WIP for #2858

### CODE REVIEW

File: docs/api/index-api.rst [hunk 1]
```diff
@@ -28,6 +28,9 @@ Submodules
     * - :mod:`models. <datatable.models>`
       - A small set of data analysis tools.
 
+    * - :mod:`time. <datatable.time>`
```
Comment (oleksiyskononenko) [inline docs/api/index-api.rst:31]: Why not `datatable.datetime`?

File: docs/api/time.rst [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+datatable.time
+==============
+
+.. list-table::
+   :widths: auto
+   :class: api-table
+
+   * - :func:`ymd(y,m,d)`
+     - Create a ``date32`` column from year, month, day components.
```
Comment (oleksiyskononenko) [inline docs/api/time.rst:12]: "year, month, day" -> "year, month and day"?

File: src/core/expr/time/fexpr_ymd.cc [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+      return Column(new YMD_ColumnImpl(
+          std::move(ycol), std::move(mcol), std::move(dcol)));
+    }
+};
+
+
+
+
+//------------------------------------------------------------------------------
+// Python-facing `qcut()` function
```
Comment (oleksiyskononenko) [inline src/core/expr/time/fexpr_ymd.cc:169]: "qcut()" -> "ymd()"

**Review by oleksiyskononenko — APPROVED** [review]

File: docs/api/index-api.rst (hunk 1 above)
Comment (st-pasha) [reply docs/api/index-api.rst:31]: I use the word "time" not in the  sense of "time of day", but as the physical _time_. That thing that keeps everything from happening at once. The fourth dimension.

For me, from a physicist's standpoint, the word "time" is perfectly self-sufficient, and doesn't need to be contorted into a "datetime" or "timestamp" or "time_point" or whatever. Specifically, a "date" is a subset of time -- 24-hour long slices of time to be precise, and the module name `dt.time` is thus suitable for describing everything temporal.

Plus, it's shorter and easier to type.

File: src/core/expr/time/fexpr_ymd.cc (hunk 3 above)
Comment (st-pasha) [reply src/core/expr/time/fexpr_ymd.cc:169]: copy-paste... :)

File: docs/api/time.rst (hunk 2 above)
Comment (st-pasha) [reply docs/api/time.rst]: ok

**Pull request merged**